main.go

```diff
@@ -103,11 +108,20 @@ func (s *Server) addUser(w http.ResponseWriter, req *http.Request) {
 }
 
 func (s *Server) handler(w http.ResponseWriter, req *http.Request) {
-	switch strings.Count(req.URL.Path, "/") { // count slash
+	switch strings.Count(req.URL.Path, "/") {
 	case 3:
 		s.processBonus(w, req)
 	case 2:
-		s.getUser(w, req)
+		if req.Method == http.MethodGet {
+			s.getUser(w, req)
+			return
+		}
+		if req.Method == http.MethodDelete {
+			s.deleteUser(w, req)
+			return
+		}
+		http.NotFound(w, req)
```

Comment:
can be simplified with either fallthrough, or by removing default and moving notFound out of switch case (preferred).